Issues during QGIS compilation

Hi @elemoine,

I have the next error when I'm building deb packages according to the command line described in the README:

```
$ docker run -it --rm -v $(pwd):/qgis -u $(id -u):$(id -g) qgis-build
...
...
...
ccache: error: Failed to create directory //.ccache/tmp: Permission denied
[6/4070] Generating qgsversion.h, qgsversion.inc
Files "/qgis/QGIS/debian/build/qgsversion.h.temp" to "/qgis/QGIS/debian/build/qgsversion.h" are different.
[7/4070] Building /qgis/QGIS/debian/build/doc/INSTALL from /qgis/QGIS/doc/INSTALL.t2t
txt2tags wrote /qgis/QGIS/debian/build/doc/INSTALL
[8/4070] Building /qgis/QGIS/debian/build/doc/INSTALL.html from /qgis/QGIS/doc/INSTALL.t2t
txt2tags wrote /qgis/QGIS/debian/build/doc/INSTALL.html
[9/4070] Building /qgis/QGIS/debian/build/doc/news from /qgis/QGIS/doc/news.t2t
txt2tags wrote /qgis/QGIS/debian/build/doc/news
[10/4070] Building /qgis/QGIS/debian/build/doc/news.html from /qgis/QGIS/doc/news.t2t
txt2tags wrote /qgis/QGIS/debian/build/doc/news.html
ninja: build stopped: subcommand failed.
debian/rules:238: recipe for target 'override_dh_auto_build-arch' failed
make[2]: *** [override_dh_auto_build-arch] Error 1
make[2]: Leaving directory '/qgis/QGIS'
debian/rules:221: recipe for target 'build-arch' failed
make[1]: *** [build-arch] Error 2
make[1]: Leaving directory '/qgis/QGIS'
debian/rules:221: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

I had to mount some volumes to make it work:

```
$ docker run -it --rm -v $(pwd):/qgis -u $(id -u):$(id -g) \
-v /etc/passwd:/etc/passwd:ro \ 
-v /etc/group:/etc/group:ro \
-v $HOME/.ccache/:$HOME/.ccache/:rw \
qgis-build
```

--- a/qgis-build/README.md
+++ b/qgis-build/README.md
@@ -33,7 +33,7 @@ $ git clone https://github.com/qgis/QGIS.git
 Build QGIS:
 
 ```shell
-$ docker run -it --rm -v $(pwd):/qgis -u $(id -u):$(id -g) qgis-build
+$ docker run -it --rm -v $(pwd):/qgis -v $HOME/.ccache:/.ccache -u $(id -u):$(id -g) -e CCACHE_DIR=/.ccache qgis-build
 ```
 
 Note that the `docker run` command should be run from the `build` directory, that is the directory
@@ -42,6 +42,8 @@ that contains the `QGIS` repository.
 Notes:
 
 * `-v $(pwd):/qgis` is used to mount the current directory (`build`) into `/qgis` in the container
+* `-v $HOME/.ccache:/.ccache` is used to mount the user's ccache directory into `/.ccache` in the container
 * `-u $(id -u):$(id -g)` is used to use the local user id and group id within the container
+* `-e CCACHE_DIR=/.ccache` is used to set the `CCACHE_DIR` envvar to `/.ccache` for the compilation
 
 After the build completes you should find Debian packages (`.deb` files) in the `build` directory.

